flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/FineGrainedSlotManagerTest.java

```diff
@@ -737,6 +739,73 @@ public void testMaxTotalResourceCpuExceeded() throws Exception {
         testMaxTotalResourceExceededRegisterResource(maxTotalResourceSetter);
     }
 
+    @Test
+    public void testGetResourceOverview() throws Exception {
+        final TaskExecutorGateway taskExecutorGateway =
+                new TestingTaskExecutorGatewayBuilder().createTestingTaskExecutorGateway();
+        final ResourceID resourceId1 = ResourceID.generate();
+        final ResourceID resourceId2 = ResourceID.generate();
+        final TaskExecutorConnection taskManagerConnection1 =
+                new TaskExecutorConnection(resourceId1, taskExecutorGateway);
+        final TaskExecutorConnection taskManagerConnection2 =
+                new TaskExecutorConnection(resourceId2, taskExecutorGateway);
+
+        final SlotID slotId1 = new SlotID(resourceId1, 0);
+        final SlotID slotId2 = new SlotID(resourceId1, 1);
+        final SlotID slotId3 = new SlotID(resourceId2, 0);
+        final SlotID slotId4 = new SlotID(resourceId2, 1);
+        final ResourceProfile resourceProfile1 = ResourceProfile.fromResources(1, 10);
+        final ResourceProfile resourceProfile2 = ResourceProfile.fromResources(2, 20);
+        final SlotStatus slotStatus1 = new SlotStatus(slotId1, resourceProfile1);
```

Comment:
There is no need to add non-allocated slot status here.

Reply:
Sry, I still did not get the point about the non-allocated slot status. I thought we should test both retrieving free resources and registered resource.
I noticed that there were several errors in my codes that lead to `getFreeResource` executed before `registerTaskManager`. Those were corrected in the new commit.

Reply:
In `FineGrainedSlotManager`, the non-allocated slot status will be ignored. It does not infect the calculation of free resources and registered resource.

Reply:
Thx. I've corrected those test cases.